bazel: fix microbenchmarks

rockwotj opened the pull request:
Do this by leveraging runfiles instead of environment variables, which
break when you switch directories. Care must be taken to keep
compatibility between CMake and Bazel, so wasm need to fall back to the
current directory for Wasm binaries, and OpenSSL tests are quite tricky
to get right with all the subtle usages of environment variables.

Also switch bench wrapper to python because the bash is becoming
non-trivial and we need to resolve some environment variables for
sanitizer configuration options.

## Backports Required

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none

rockwotj committed:
bazel: fix micro benchmarks

Do this by leveraging runfiles instead of environment variables, which
break when you switch directories. Care must be taken to keep
compatibility between CMake and Bazel, so wasm need to fall back to the
current directory for Wasm binaries, and OpenSSL tests are quite tricky
to get right with all the subtle usages of environment variables.

Also switch bench wrapper to python because the bash is becoming
non-trivial and we need to resolve some environment variables for
sanitizer configuration options.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--fec693e4-4bf2-404a-a2bc-8e36637aa707-->
<!--TEST RESULTS-->
<!--5ba124bb-4808-409e-aa34-4f39e77c80d8-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/64450">build#64450</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.preparing.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/64450#01962831-0d62-4d8e-bcc9-f75fb2759601 | FLAKY | 1/4 |
| rptest.tests.upgrade_test.UpgradeBackToBackTest.test_upgrade_with_all_workloads.single_upgrade=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/64450#01962831-0d60-4063-ba46-58b5b7dd3ad8 | FLAKY | 1/4 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/64452">build#64452</a></summary>


| test_id                                                                                                                                                   | test_kind   | job_url                                                                                   | test_status   | passed   |
|:----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.controller_log_limiting_test.ControllerLogLimitPartitionBalancerTests.test_partition_balancer_with_limits                                    | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64452#019629f1-9b98-4e36-80bd-850e50ed74f5 | FLAKY         | 1/2      |
| rptest.tests.datalake.throttling_test.DatalakeThrottlingTest.test_basic_throttling.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64452#019629f1-9b97-4e5c-b75e-116be7bca440 | FLAKY         | 1/3      |
| rptest.tests.scaling_up_test.ScalingUpTest.test_topic_hot_spots                                                                                           | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64452#019629ec-5506-4d2d-ab49-a9e465962960 | FLAKY         | 1/2      |
| rptest.tests.tiered_storage_pause_test.TestTieredStoragePause.test_safe_pause_resume.allow_gaps_topic_level=True.allow_gaps_cluster_level=False           | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64452#019629f1-9b98-4e36-80bd-850e50ed74f5 | FLAKY         | 1/2      |
</details>
<!--/body-->

rockwotj committed:
bazel: use script bootstrap for python binaries/tools

This prevents us from needing to use the explicit $(PYTHON3) trick in
genrule, and also fixes `py_test` targets in CI, which doesn't have a
system interpreter with the name `python3`.

StephanDollberg reviewed (commented):
Looks good to me but should probably wait for Travis & Mike

michael-redpanda commented on bazel/bench_wrapper.py:
```diff
@@ -0,0 +1,92 @@
+                v = str(Path(v).resolve())
+            resolved_options.append(f"{k}={v}")
+        os.environ[opt_name] = ":".join(resolved_options)
+    # This is sort of horrible, but this sets the include search directory for openssl configuration file.
+    # In CMake we used an absolute path for the .include (which is what OpenSSL recommends), but since
+    # Bazel tries *really* hard to use relative paths for better caching, we need to add the current path
+    # to the include path for the openssl configuration.
+    # See: https://docs.openssl.org/3.1/man5/config/#environment
+    os.environ["OPENSSL_CONF_INCLUDE"] = str(Path().resolve())
```
🫠

rockwotj replied:
Open to other options!

michael-redpanda replied:
No shade on you... definitely is a pita to deal with :)

michael-redpanda reviewed (approved):
Thanks, changes to OpenSSL look good.  Appreciate you doing this.

StephanDollberg commented:
looks like the fips mb is still broken https://buildkite.com/redpanda/redpanda/builds/64516#01963693-8774-4a27-b54d-a56c19110772

rockwotj commented:
that is confusing because we should be doing the same thing in the tests, but thanks for flagging. I will look at it
